codecs/text/src/test/java/com/datastax/oss/dsbulk/codecs/text/string/StringToVectorCodecTest.java

```diff
@@ -53,39 +72,46 @@ void should_convert_from_valid_external() {
 
   @Test
   void should_convert_from_valid_internal() {
-    assertThat(dsbulkCodec)
+    assertThat(codec)
         .convertsFromInternal(vector)
-        .toExternal(vectorCodec.format(vector))
+        .toExternal(
+            "[1.1,2.2,3.3,4.4,5.5]") // this is NOT 100% identical to vector CQL representation
         .convertsFromInternal(null)
         .toExternal("NULL");
-
-    // We should encode
   }
 
   @Test
-  void should_not_convert_from_invalid_internal() {
-    assertThat(dsbulkCodec).cannotConvertFromInternal("not a valid vector");
+  void should_not_convert_from_invalid_external() {
+    assertThat(codec).cannotConvertFromExternal("[6.646329843]");
   }
 
   // To keep usage consistent with VectorCodec we confirm that we support encoding when too many
-  // elements are
-  // available but not when too few are.  Note that it's actually VectorCodec that enforces this
-  // constraint so we
-  // have to go through encode() rather than the internal/external methods.
+  // elements are available but not when too few are.  Note that it's actually VectorCodec that
+  // enforces this constraint so we have to go through encode() rather than the internal/external
+  // methods.
   @Test
   void should_encode_too_many_but_not_too_few() {
 
     ArrayList<Float> tooMany = Lists.newArrayList(values);
     tooMany.add(6.6f);
     CqlVector<Float> tooManyVector = CqlVector.newInstance(tooMany);
-    String tooManyString = dsbulkCodec.internalToExternal(tooManyVector);
+    String tooManyString = codec.internalToExternal(tooManyVector);
     ArrayList<Float> tooFew = Lists.newArrayList(values);
     tooFew.remove(0);
     CqlVector<Float> tooFewVector = CqlVector.newInstance(tooFew);
-    String tooFewString = dsbulkCodec.internalToExternal(tooFewVector);
+    String tooFewString = codec.internalToExternal(tooFewVector);
 
-    assertThat(dsbulkCodec.encode(tooManyString, ProtocolVersion.DEFAULT)).isNotNull();
-    assertThatThrownBy(() -> dsbulkCodec.encode(tooFewString, ProtocolVersion.DEFAULT))
+    assertThat(codec.encode(tooManyString, ProtocolVersion.DEFAULT)).isNotNull();
+    assertThatThrownBy(() -> codec.encode(tooFewString, ProtocolVersion.DEFAULT))
         .isInstanceOf(IllegalArgumentException.class);
   }
+
+  // Issue 484: now that we're using the dsbulk string-to-subtype converters we should get
+  // enforcement of existing dsbulk policies.  For our purposes that means the failure on
+  // arithmetic overflow.
+  @Test
+  void should_not_convert_too_much_precision() {
+    assertThatThrownBy(() -> codec.encode("6.646329843", ProtocolVersion.DEFAULT))
```

Comment:
I'd argue this is incorrect.  This test is intended to mirror the equivalent test in JsonNodeToVectorCodecTest.  In that case we're trying to confirm that the JSON representation for an otherwise valid vector (really just a JSON array in that case) fails to convert because precision policies in the dsbulk codecs are being enforced.  If we want to model the same thing here this should be the string representation for an otherwise valid vector... which means it should be something like:

```java
  // Issue 484: now that we're using the dsbulk string-to-subtype converters we should get
  // enforcement of existing dsbulk policies.  For our purposes that means the failure on
  // arithmetic overflow.
  @Test
  void should_not_convert_too_much_precision() {
    assertThatThrownBy(() -> codec.encode("[1.1, 2.2, 3.3, 6.646329843]", ProtocolVersion.DEFAULT))
        .isInstanceOf(ArithmeticException.class);
  }
```

Reply:
Your example if fine, but the current one is too. In DSBulk, enclosing brackets and braces are generally optional. So `codec.encode("6.646329843", ProtocolVersion.DEFAULT))` should behave like `codec.encode("[6.646329843]", ProtocolVersion.DEFAULT))`. You can add both tests btw.

But note that this test is almost identical to `should_not_convert_from_invalid_external`. Maybe merge both into one single test?